tests/trainer/test_dataloaders.py

```diff
@@ -1159,3 +1159,72 @@ def test_replace_sampler_with_multiprocessing_context(tmpdir):
 
     new_data_loader = trainer.replace_sampler(train, SequentialSampler(train.dataset))
     assert (new_data_loader.multiprocessing_context == train.multiprocessing_context)
+
+
+@pytest.mark.parametrize('multiple_trainloader_mode', ["min_size", "max_size_cycle"])
+def test_correct_dataloader_idx_in_hooks(tmpdir, multiple_trainloader_mode):
+    """
+    Check the correct dataloader_idx inside hooks
+    """
+
+    class CustomEvalModelTemplate(EvalModelTemplate):
```

Comment:
Can you inherit from `BoringModel` instead?

Reply:
Yeah more work 😅

Reply:
If it's too much, I don't mind leaving it as is. But we want to avoid adding new tests relying on EvalModelTemplate as we want to eventually remove it

Reply:
ok done now :)